app/services/migration/migrators/contract.rb

```diff
@@ -28,25 +28,25 @@ def ecf_contracts
       end
 
       def dependencies
-        %i[cohort lead_provider course statement]
+        %i[course statement]
       end
     end
 
     def call
       migrate(self.class.ecf_contracts) do |ecf_contract|
-        course = courses_by_identifier[ecf_contract.course_identifier]
+        course_id = find_course_id!(identifier: ecf_contract.course_identifier)
 
         ecf_statements(ecf_contract).find_each do |ecf_statement|
-          statement = ::Statement.find_by!(ecf_id: ecf_statement.id)
+          statement_id = find_statement_id!(ecf_id: ecf_statement.id)
 
           contract_template = ::ContractTemplate.find_or_initialize_by(ecf_id: ecf_contract.id)
-          contract_template.update!(ecf_contract.attributes.slice(SHARED_ATTRIBUTES))
+          contract_template.update!(ecf_contract.attributes.slice(*SHARED_ATTRIBUTES))
 
-          contract = ::Contract.find_or_initialize_by(
-            statement:,
```

Comment:
I think we should keep this the same as it was as we should always have one unique entry for statement/course